src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -162,5 +162,7 @@ sealed class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering:
       iterator.foreach { case (key, value) => update(key, f(key, value)) }
       this
     }
+
+    override def clone() = super.clone().rangeImpl(from, until)
```

Comment:
I am not sure this is the right behavior, if I'm understanding this correctly.  If you have a small view of a large underlying collection, do you really want to clone the entire underlying collection?  I think, instead, what you probably mean is "I want a copy of this data".  If you want to preserve the viewness of it (which I guess is necessary for types), and you don't want a second view of the same collection, you should slice the underlying collection and view that from end to end.  But other views (e.g. `ArrayBuffer`) when cloned just access the same underlying data.